lib/Utils/Layout/Codegen.cpp

```diff
@@ -381,6 +381,9 @@ FailureOr<scf::ForOp> MLIRLoopNestGenerator::generateForLoop(
   isl_ast_expr_free(expr);
   isl_ast_node_free(node);
 
+  // Restore the insertion point to after the outermost loop.
```

Comment:
would setting an insertion guard at the start of the func work?

Reply:
Yes! I forgot about that.